applications/GeoMechanicsApplication/custom_elements/triangle_2d_15.hpp

```diff
@@ -0,0 +1,1752 @@
+            rResult[14](1, 0) = -384.0 * et2 + (-512.0 * xi1 + 320.0) * et1 + 64.0 * xi1 - 32.0;
+            rResult[14](1, 1) = -64.0 * xi1 * (4.0 * xi1 + 12.0 * et1 - 5.0);
+            //
+            return rResult;
+        }
+
+        // ========================================================================================
+        // ========================================================================================
+        /**
+         * returns the third order derivatives of all shape functions
+         * in given arbitrary pointers
+         * @param rResult a fourth order tensor which contains the third derivatives
+         * @param rPoint the given point the third order derivatives are calculated in
+         */
+        ShapeFunctionsThirdDerivativesType& ShapeFunctionsThirdDerivatives(ShapeFunctionsThirdDerivativesType& rResult, const CoordinatesArrayType& rPoint) const override
+        {
+            if (rResult.size() != this->PointsNumber())
+            {
+                ShapeFunctionsThirdDerivativesType temp(this->PointsNumber());
+                rResult.swap(temp);
+            }
+            for (IndexType i = 0; i < rResult.size(); i++)
+            {
+                DenseVector<Matrix> temp(this->PointsNumber());
+                rResult[i].swap(temp);
+            }
+            //
+            rResult[0][0].resize(2, 2, false);
+            rResult[0][1].resize(2, 2, false);
+            rResult[1][0].resize(2, 2, false);
+            rResult[1][1].resize(2, 2, false);
+            rResult[2][0].resize(2, 2, false);
+            rResult[2][1].resize(2, 2, false);
+            rResult[3][0].resize(2, 2, false);
+            rResult[3][1].resize(2, 2, false);
+            rResult[4][0].resize(2, 2, false);
+            rResult[4][1].resize(2, 2, false);
+            rResult[5][0].resize(2, 2, false);
+            rResult[5][1].resize(2, 2, false);
+            rResult[6][0].resize(2, 2, false);
+            rResult[6][1].resize(2, 2, false);
+            rResult[7][0].resize(2, 2, false);
+            rResult[7][1].resize(2, 2, false);
+            rResult[8][0].resize(2, 2, false);
+            rResult[8][1].resize(2, 2, false);
+            rResult[9][0].resize(2, 2, false);
+            rResult[9][1].resize(2, 2, false);
+            rResult[10][0].resize(2, 2, false);
+            rResult[10][1].resize(2, 2, false);
+            rResult[11][0].resize(2, 2, false);
+            rResult[11][1].resize(2, 2, false);
+            rResult[12][0].resize(2, 2, false);
+            rResult[12][1].resize(2, 2, false);
+            rResult[13][0].resize(2, 2, false);
+            rResult[13][1].resize(2, 2, false);
+            rResult[14][0].resize(2, 2, false);
+            rResult[14][1].resize(2, 2, false);
+            //
+            double fx3 = 256.0 * rPoint[0] + 256.0 * rPoint[1] - 160.0;
+            double fx2y = fx3;
```

Comment:
const?

Reply:
These variables are re-used in the same method (overwritten). They are not constants.